src/jpeg.imageio/jpegoutput.cpp

```diff
@@ -185,6 +185,12 @@ JpgOutput::open(const std::string& name, const ImageSpec& newspec,
         }
         DBG std::cout << "out open: set_colorspace\n";
 
+        // Save as a progressive jpeg if requested by the user
+        ParamValue* progressive = m_spec.find_attribute("jpeg:progressive", TypeDesc::INT);
+        if (progressive && *(int*)progressive->data() != 0) {
```

Comment:
There's a new idiom that is a more compact way to express this!

    if (m_spec.get_int_attribute("jpeg:progressive")) {

(The value it returns if the attrib doesn't exist is 0, so this works out perfectly in this case. Though you can also pass a second argument that is the default to use if there is no attribute.)

Or, since 2.1, this also works (which style you prefer might perhaps be a matter of personal taste):

     if (m_spec["jpeg:progressive"].get<int>()) {

Reply:
Nice, thank you. I went with the first option as it looks a little cleaner to me.